sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/PythonUDF.scala

```diff
@@ -67,4 +67,12 @@ case class PythonUDF(
     exprId = resultId)
 
   override def nullable: Boolean = true
+
+  /**
+   * `resultId` can be seen as cosmetic variation in PythonUDF, as it doesn't affect the result.
```

Comment:
no biggie but maybe it's better to put this comment around  `Canonicalize.execute(this.copy(resultId = ExprId(-1))` (not as a method doc).

Reply:
oh, right.